packages/synapse-react-client/src/components/SetAccessRequirementCommonFields/SetAccessRequirementCommonFields.tsx

```diff
@@ -199,26 +211,30 @@ export const SetAccessRequirementCommonFields = React.forwardRef(
             const isStillLoading =
               (isEditing && !accessRequirement) || !subjectsType
             const hasSubjectsError = hasPendingSubjects || subjects.length === 0
-
-            if (isStillLoading || hasSubjectsError) {
-              if (hasSubjectsError && !isStillLoading) {
-                if (hasPendingSubjects) {
-                  setSubjectsError(UNSAVED_SUBJECTS_ERROR_MESSAGE(subjectsType))
-                } else if (subjects.length === 0) {
-                  setSubjectsError(EMPTY_SUBJECT_LIST_ERROR_MESSAGE)
+            if (!subjectsDefinedByAnnotations) {
+              if (isStillLoading || hasSubjectsError) {
```

Comment:
When the parent calls `save()`, if this component `isStillLoading`, then this component should call `onError` to inform the parent that the AR wasn't saved, so I think we may need to remove the outer check for `!subjectsDefinedByAnnotations`.

What do you think about changing `hasSubjectsError` to include a check for `subjectsDefinedByAnnotations`, so that it's clearer that there can only be a subjects error when the AR's subjects aren't defined by annotations? Something like -- 

```ts
const hasSubjectsError = !subjectsDefinedByAnnotations && (hasPendingSubjects || subjects.length === 0)
if (isStillLoading || hasSubjectsError) {
  ... 
  onError()
  return
}
```

Reply:
Oops, made this change with the intent of revisiting later.  Made your suggested change to the logic